optuna/integration/skopt.py

```diff
@@ -303,20 +293,11 @@ def ask(self) -> Dict[str, Any]:
         params = {}
         param_values = self._optimizer.ask()
         for (name, distribution), value in zip(sorted(self._search_space.items()), param_values):
-            if isinstance(
-                distribution,
-                (
-                    distributions.UniformDistribution,
-                    distributions.LogUniformDistribution,
-                    distributions.DiscreteUniformDistribution,
-                    distributions.FloatDistribution,
-                ),
-            ):
+            if isinstance(distribution, distributions.FloatDistribution):
                 # Type of value is np.floating, so cast it to Python's built-in float.
                 value = float(value)
-            if isinstance(distribution, distributions.DiscreteUniformDistribution):
-                value = value * distribution.q + distribution.low
-            elif isinstance(distribution, distributions.FloatDistribution):
+
+            if isinstance(distribution, distributions.FloatDistribution):
```

Comment:
Now we can simply remove this `if` statement, I think (thanks to the cleanup).

before:

```python
            if isinstance(distribution, distributions.FloatDistribution):
                # Type of value is np.floating, so cast it to Python's built-in float.
                value = float(value)

            if isinstance(distribution, distributions.FloatDistribution):
                if distribution.step is not None:
                    value = value * distribution.step + distribution.low
            elif isinstance(distribution, distributions.IntDistribution):
                if distribution.log:
                    value = int(np.round(value))
                    value = min(max(value, distribution.low), distribution.high)
                else:
                    value = int(value * distribution.step + distribution.low)
```

after:

```python
            if isinstance(distribution, distributions.FloatDistribution):
                # Type of value is np.floating, so cast it to Python's built-in float.
                value = float(value)
                if distribution.step is not None:
                    value = value * distribution.step + distribution.low

            elif isinstance(distribution, distributions.IntDistribution):
                if distribution.log:
                    value = int(np.round(value))
                    value = min(max(value, distribution.low), distribution.high)
                else:
                    value = int(value * distribution.step + distribution.low)
```

Reply:
Done in 06d6c8d5ffc729d510e15cb7493a8b14e7c07fce!